numpyro/distributions/util.py

```diff
@@ -347,41 +347,50 @@ def clamp_probs(probs):
 
 
 def von_mises_centered(key, concentration, shape=(), dtype=jnp.float64):
-    return _von_mises_centered(key, concentration, shape, dtype)
-
-
-@partial(jit, static_argnums=(2, 3))
-def _von_mises_centered(key, concentration, shape, dtype):
     """ Compute centered von Mises samples using rejection sampling from [1] with wrapped Cauchy proposal.
 
         *** References ***
         [1] Luc Devroye "Non-Uniform Random Variate Generation", Springer-Verlag, 1986;
             Chapter 9, p. 473-476. http://www.nrbook.com/devroye/Devroye_files/chapter_nine.pdf
 
-        :param sample_shape: shape of samples
-        :param rng_key: random number generator key
+
+        :param key: random number generator key
+        :param concentration: concentration of distribution
+        :param shape: shape of samples
+        :param dtype: float precesions for choosing correct s cutfoff
         :return: centered samples from von Mises
     """
+    shape = shape or jnp.shape(concentration)
+    dtype = canonicalize_dtype(dtype)
+    concentration = lax.convert_element_type(concentration, dtype)
+    concentration = jnp.broadcast_to(concentration, shape)
+    return _von_mises_centered(key, concentration, shape, dtype)
 
+
+@partial(jit, static_argnums=(2, 3))
+def _von_mises_centered(key, concentration, shape, dtype, max_iter=100):
```

Comment:
nit: maybe it is better to hard-code `max_iter` inside the function. But it is fine to leave it as-is if you intend to change it for your usage. If you want to keep, you can add `max_iter` keyword to `von_mises_centered` too.

Reply:
I'll hard-code it.